### ISSUE TITLE
Possibility to define barrel file

### ISSUE THREAD

**Issue by rainerhahnekamp** [softarc-consulting/sheriff#36]
`index.ts` is the name of the file representing a module.

It should be possible to go with another one. For example, the Angular CLI calls it `public-api.ts`. 

@mikezks: You want to give it a try?

**Comment by bradley-carestia-jbh** [comment]
Would love to see this fixed to allow Sherriff to handle cross-entry-point dependencies without having to rename all my public-api files

**Comment by rainerhahnekamp** [comment]
@KvikullSigurdur just to avoid a misunderstanding. You can have both barrel modules (with an `index.tx`) and barrel-less in a project.

If you now say that those barrel modules are breaking encapsulation the whole time, then I'd first try to fix that.

---

That said, I think i don't quite catch the idea of the `barrelIndex.ts`. What is that one supposed to do? What that be placed next to the `index.ts`and be used for the "real encapsulation"?

If yes, `sheriff.config.ts` has a property `barrelFileName`. Set that one to `barrelIndex.ts` and Sheriff would ignore `index.ts`.

Let me know if that works for you

**Comment by rainerhahnekamp** [comment]
@bradley-carestia-jbh, same goes for you. Just set `barrelFileName` in `sheriff.config.ts` to `public-api.ts` and you're done.

**Comment by KvikullSigurdur** [comment]
> If you now say that those barrel modules are breaking encapsulation the whole time, then I'd first try to fix that.

Yep that is the goal :) Jump wanted to know if I could somehow enable the warning incrimentally. barrelFileName works perfectly. Thank you very much!

**Comment by bradley-carestia-jbh** [comment]
Awesome, I'll give that a try next time I'm looking at this.
